backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java

```diff
@@ -268,4 +273,53 @@ public Element dropRequirementSubjectFromDiagram(Element droppedElement, Node dr
         }
         return droppedElement;
     }
+
+    public Element dropElementFromDiagramInRequirementAssumeConstraintCompartment(Element droppedElement, Node droppedNode, Element targetElement, Node targetNode, IEditingContext editingContext, IDiagramContext diagramContext,
+            Map<org.eclipse.sirius.components.view.diagram.NodeDescription, NodeDescription> convertedNodes) {
+        if (droppedElement instanceof ConstraintUsage droppedConstraint) {
+            if (targetElement instanceof RequirementUsage requirementUsage) {
```

Comment:
Could you merge the two `if` in one please?

Reply:
Done.